### PR TITLE
update preview panel to align with latest ux

### PR DESCRIPTION
**Pull request by christianvogt**
https://issues.redhat.com/browse/RHOAIENG-11539

## Description

Updated the preview header to use a white background and made it sticky to the top so that the scrollable area is the drawer body.

No connection type name or description:
image
 
image

Has connection type name, description and category:
image
image
image

Popovers for field descriptions:
image

Updated dropdown review to include values:
image

Preview hidden default read only values. Use popover to see real value.
image
image

cc @simrandhaliw 

Additional changes:
- update the config map types to include category
- updated serialization utils to handle category
- added reusable `CategoryLabel` component
- adjusted field table column widths
- used `React.memo` for performance improvements

## How Has This Been Tested?

Existing tests. Manually. For the most part this is just look and feel changes.

- Enable the connection types feature flag
- Go to Settings -> Connection types
- Create a connection type
- Open the preview panel
- Create various fields with different values

## Test Impact

N/A

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/concepts/connectionTypes/ConnectionTypePreview.tsx [hunk 1]
```diff
@@ -1,44 +1,113 @@
...
+// TODO consider refactoring this form for reuse when creating connection type instances
+const ConnectionTypePreview: React.FC<Props> = ({ obj }) => {
+  const connectionTypeName = obj && obj.metadata.annotations?.['openshift.io/display-name'];
+  const connectionTypeDescription = (obj && getDescriptionFromK8sResource(obj)) ?? undefined;
+  return (
+    <Form>
+      <Title headingLevel="h1">Add connection</Title>
+      <FormGroup label="Connection type" fieldId="connection-type" isRequired>
+        <SimpleSelect
+          id="connectio-type"
```
Comment (emilys314) [inline frontend/src/concepts/connectionTypes/ConnectionTypePreview.tsx:40]: "connectio-type" typo 🙂

**Comment by christianvogt** [comment]
Rebased and fixed up unit tests 🤞

**Review by jeff-phillips-18 — APPROVED** [review]

**Comment by christianvogt** [comment]
Going to need https://github.com/opendatahub-io/odh-dashboard/pull/3119 fixed it seems to get around this test failure.

**Comment by christianvogt** [comment]
rebased

**Pull request merged**